pulp_service/pulp_service/app/tasks/ansible_log_parser.py

```diff
@@ -0,0 +1,229 @@
+import re
+import copy
+import logging
+import aiohttp
+from asgiref.sync import sync_to_async
+from pulp_service.app.models import AnsibleLogReport
+from pulpcore.app.models import Domain
+from pulpcore.app.models import CreatedResource
+from pulpcore.plugin.tasking import dispatch
+
+_logger = logging.getLogger(__name__)
+
+# Define regular expressions for log parsing (keep these as they are)
+SYSTEM_ROLE_LOG_RE = re.compile(
+    r"/SYSTEM-ROLE-(?P<role>[a-z0-9_]+)_(?P<test_name>tests_[a-z0-9_]+"
+    r"[.]yml)-.*-ANSIBLE-(?P<ansible_ver>[0-9.]+).*[.]log$"
+)
+SYSTEM_ROLE_TF_LOG_RE = re.compile(
+    r"/data/(?P<role>[a-z0-9_]+)-(?P<test_name>tests_[a-z0-9_]+)"
+    r"-ANSIBLE-(?P<ansible_ver>[0-9.]+)-(?P<tf_job_name>[0-9a-z_]+)"
+    r"-(?P<test_status>SUCCESS|FAIL)[.]log$"
+)
+
+
+async def get_file_data(log_url, timeout=30):
+    """Fetch file content from URL asynchronously."""
+    try:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(log_url, timeout=timeout) as response:
+                response.raise_for_status()
+                text = await response.text()
+                return text.splitlines()
+    except aiohttp.ClientError as e:
+        _logger.error("Failed to fetch log file: %s", str(e))
+        raise ValueError(f"Error fetching log file: {str(e)}")
+
+
+async def analyze_ansible_log(log_url, role_filter, domain_id=None):
```

Comment:
With the default being specified in the dispatch function I recommend not having it be optional here, but clarifying in the docstring it can be UUID or None.